File: src/co/edu/uptc/music/logic/managers/GenresManager.java
package co.edu.uptc.music.logic.managers;

import java.sql.ResultSet;

import co.edu.uptc.music.logic.models.Genre;
import co.edu.uptc.music.persistence.GenreDAO;

public class GenresManager extends BaseManager<Genre> {

    private GenreDAO genreDAO;

    public GenresManager() {
        genreDAO = new GenreDAO();
    }

    @Override
    public Genre findItem(String s) {
        for (Genre genre : getList()) {
            if (genre.getDescription().equalsIgnoreCase(s)) return genre;
        }
        return null;
    }

    private void internalLoad(ResultSet res) {
        if (res != null) {
            try {
                clearList();
                while (res.next()) {
                    String genId = res.getString("GENRE_ID");
                    String desc = res.getString("DESCRIPTION");
                    String imgPath = res.getString("IMG_PATH");
                    addItem(new Genre(genId, desc, imgPath));
                }
            } catch (Exception ignored) {
            }
        }
    }

    public void loadGenres() {
        internalLoad(genreDAO.queryGenres());
    }

}

File: src/co/soundwave/managers/GenresManager.kt
package co.soundwave.managers

import co.soundwave.extensions.ignore
import co.soundwave.models.Genre
import co.soundwave.repository.genre.GenreDAO

class GenresManager : BaseManager<GenreDAO, Genre>() {
    
    override val dao: GenreDAO = GenreDAO()
    
    override fun findItem(id: Int): Genre? = getList().firstOrNull { it.id == id }
    
    override fun findItem(s: String): Genre? = getList().firstOrNull { it.name.equals(s, true) }
    
    override fun load() {
        dao.query()?.let {
            ignore {
                clearList()
                while (it.next()) {
                    val genId = it.getInt("id_${dao.sql.tableName()}")
                    val desc = it.getString("name_${dao.sql.tableName()}")
                    val imgPath = it.getString("img_path_${dao.sql.tableName()}")
                    addItem(Genre(genId, desc, imgPath))
                }
            }
        }
    }
}